### DISCUSSION TITLE
facing problem with a diesel conditional query

### DISCUSSION THREAD

**Discussion by jiangxiaoqiang** [diesel-rs/diesel#3091] [Q&A]
I am now work with a diesel conditional query, facing a problem like this:

https://stackoverflow.com/questions/71535350/function-or-associated-item-not-found-in-dolphin-schemainterviewtable

I did not know what should I do to make it work, anyone facing the similar problem? any suggestion is appreciate! I made a minimal reproduce and pasted in the stackoverflow.

I have made a minimal reproduce:

`git clone -b diesel-condition-query https://github.com/jiangxiaoqiang/rust-learn.git
`

**Comment by weiznich** [comment]
Please always provide the following information if you ask questions:

* A reproducible example of the code causing your problems
* A description of what you want to do
* A description of how that fails given the reproducible example
* The error message emitted by rustc, if the compilation fails
* General information about your setup:
    * Rustc version
    * Diesel version + feature flags
    * Operating system
    * Database system

And no, just linking to some other question and stating your have the same problem does not provide these information. If you expect that other people spend time on answering your question, you need to invest time in your questions.

As it seems like I have to ask for this information everytime you ask a question: This is the last time I response to your question, if the question is missing those necessary information.

**Reply by jiangxiaoqiang** [reply]
I have tried to make a minimal reproduce to show what was happen with the diesel condition query.

**Reply by weiznich** [reply]
I've tried to build your example project and got a different error than mentioned in your question. Please clarify what exactly you want to know. Again: Don't expect other people to spend large amounts of time trying to understand what you what to know and to debug your code if it's not even matching your questions. Please spend some time to write down concrete questions including the relevant code parts + error messages. 
That's really my last message on any of your questions otherwise.

**Reply by jiangxiaoqiang** [reply]
I am very appreciated that you take time to help me, sorry for wasting your time, I finally figure out what should I do by reading the diesel source code.
